tests/integration/pilot/mcs/serviceexport_test.go

```diff
@@ -89,6 +90,38 @@ func TestServiceExports(t *testing.T) {
 		Run(func(ctx framework.TestContext) {
 			cluster := ctx.Clusters().Default()
 
+			// restarting the istiod deployment because we short-circuit the serviceexport controller if the CRD isn't there on istiod startup
```

Comment:
Can you clarify why this is needed? This sounds like a bug

Reply:
That goes back to this change: https://github.com/istio/istio/pull/29763/files#diff-dabdc9b878f95aed9dee67d29e869812e924607537416f2aa9ebf4f79ed0142fR96 which I think we discussed in one of the Intuit-MCS meetings. Essentially - we don't have a way for the controller to start or stop itself so we just permanently short circuit the logic if the CRD doesn't exist, and log a message telling the user to apply the CRD and restart istiod.

We could do some form of periodic retries if we wanted to, but that also seems like a very edge case

Reply:
can you follow the pattern used here: https://github.com/istio/istio/blob/fd6ecf7e4edbb080c31a0146c856c30ae66efb87/tests/integration/pilot/main_test.go#L55 so we don't need to restart?

Reply:
Alternatively you could install it on the clusters in the prow script before running `go test`.